test/fees/CompoundBorrowFeeCalculator.t.sol

```diff
@@ -0,0 +1,159 @@
+}
+
+contract CompoundBorrowFeeCalculatorTest is Test {
+    event FeeCharged(address indexed token, uint256 amount, bytes32 metadata);
+
+    address public constant NATIVE = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    address public constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address public constant C_USDCV3 = 0xc3d688B66703497DAA19211EEdff47f25384cdc3;
+    address public constant DUMMY_TO_ADDRESS = address(0);
+    bytes32 public constant META_DATA = bytes32(bytes('compound-v3:borrow'));
+    bytes4 public constant COMPOUND_BORROW_SELECTOR =
+        bytes4(keccak256(bytes('withdrawFrom(address,address,address,uint256)')));
+    uint256 public constant BPS_BASE = 10_000;
+    uint256 public constant SIGNER_REFERRAL = 1;
+
+    address public user;
+    address public feeCollector;
+    Router public router;
+    IAgent public userAgent;
+    address public borrowFeeCalculator;
+
+    // Empty arrays
+    address[] public tokensReturnEmpty;
+    IParam.Input[] public inputsEmpty;
+
+    address public comet = C_USDCV3;
+    address public collateral = WETH;
+    uint256 public collateralAmount = 3000 ether;
+    address public baseToken = IComet(comet).baseToken();
+
+    function setUp() external {
+        user = makeAddr('User');
+        feeCollector = makeAddr('FeeCollector');
+        address pauser = makeAddr('Pauser');
+
+        // Depoly contracts
+        router = new Router(makeAddr('WrappedNative'), pauser, feeCollector);
+        vm.prank(user);
+        userAgent = IAgent(router.newAgent());
+        borrowFeeCalculator = address(new CompoundBorrowFeeCalculator(address(router), 0));
+
+        // Setup collateral
+        vm.startPrank(user);
+        deal(collateral, user, collateralAmount);
+        IERC20(collateral).approve(comet, collateralAmount);
+        IComet(comet).supply(collateral, collateralAmount);
+        IComet(comet).allow(address(userAgent), true);
+        vm.stopPrank();
+
+        // Setup fee calculator
+        bytes4[] memory selectors = new bytes4[](1);
+        selectors[0] = COMPOUND_BORROW_SELECTOR;
+        address[] memory tos = new address[](1);
+        tos[0] = DUMMY_TO_ADDRESS;
+        address[] memory feeCalculators = new address[](1);
+        feeCalculators[0] = borrowFeeCalculator;
+        router.setFeeCalculators(selectors, tos, feeCalculators);
+
+        vm.label(address(router), 'Router');
+        vm.label(address(userAgent), 'UserAgent');
+        vm.label(feeCollector, 'FeeCollector');
+        vm.label(address(borrowFeeCalculator), 'BorrowFeeCalculator');
+        vm.label(C_USDCV3, 'cUSDCv3');
+    }
+
+    function testChargeBorrowFee(uint256 amount, uint256 feeRate) external {
+        feeRate = bound(feeRate, 0, BPS_BASE - 1);
+
+        // Calculate borrow limit, assume ETH price is 1000
+        IComet.AssetInfo memory assetInfo = IComet(comet).getAssetInfoByAddress(collateral);
+        uint256 collateralRatio = assetInfo.borrowCollateralFactor;
```

Comment:
The collateral ratiao is 1e18 format?

Reply:
yes, reference https://github.com/compound-developers/compound-3-developer-faq/blob/master/examples/get-cf-examples.js#L70

Reply:
I see. Originally I thought why divide by 1e18